lib/optimize/ConcatenatedModule.js

```diff
@@ -1113,6 +1111,8 @@ class ConcatenatedModule extends Module {
 
 		// List of all used names to avoid conflicts
 		const allUsedNames = new Set(RESERVED_NAMES);
+		// Top level declarations should be updated with respect of renaming
+		const topLevelDeclarations = this.buildInfo.topLevelDeclarations;
```

Comment:
It's not allow to change `buildInfo` during code generation.

After a single build code generation could run multiple times. Modifying it would leave weird side effects.

Instead create a new `Set` and expose that as code generation result. We have a `data` map in the code generation result where this can be places `data.set("topLevelDeclarations", topLevelDeclarations)`